src/theory/fp/fp_converter.cpp

```diff
@@ -0,0 +1,54 @@
+/*********************                                                        */
+/*! \file fp_converter.cpp
+ ** \verbatim
+ ** Original author: Martin Brain
+ ** Major contributors: 
+ ** Minor contributors (to current version): 
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2014  University of Oxford
+ ** See the file COPYING in the top-level source directory for licensing
+ ** information.\endverbatim
+ **
+ ** \brief [[ Conversion of floating-point operations to bit-vectors using symfpu. ]]
+ **
+ **/
+
+
+#include "theory/fp/fp_converter.h"
+
+#include <stack>
+
+
+namespace CVC4 {
+namespace theory {
+namespace fp {
+
+
+  size_t PairTypeNodeHashFunction::operator()(const std::pair<TypeNode, TypeNode> &p) const {
+    TypeNodeHashFunction h;
+    return h(p.first) ^ h(p.second);
```

Comment:
Lets add a function:
size_t CombineHashes(size_t a, size_t b) {
  return a ^ (b + 0x9e3779b9 + (a << 6) + (a >> 2));
}
This is basically just inlining http://www.boost.org/doc/libs/1_35_0/doc/html/boost/hash_combine_id241013.html

It is fine to just leave this anonymous for now.